.github/workflows/test.yml

```diff
@@ -28,7 +29,9 @@ jobs:
           path: ${{ env.pythonLocation }}
           key: ${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}-${{ hashFiles('requirements.dev.txt') }}
       - name: Install dependencies
-        run: pip install -U pip setuptools wheel -r requirements.dev.txt -e .
+        run: pip install -U pip setuptools wheel pre-commit -r requirements.dev.txt -e .
+      - name: pre-commit
+        run: pre-commit run --from-ref origin/main --to-ref HEAD
```

Comment:
Why not `pre-commit run --all-files`? This assumes that the PR goes to main, which is not always the case.
Per the [pre-commit docs](https://pre-commit.com/#cli):
"`pre-commit run --all-files`: run all the hooks against all the files. This is a useful invocation if you are using pre-commit in CI."